frontend/src/app/auth.service.ts

```diff
@@ -45,55 +50,69 @@ export class AuthService {
-          this.loginEvent.next({ success: true });
-          this.router.navigate(['']);
-        },
-        err => {
-          this.loginEvent.next({ success: false, message: err });
-        }
+      user => {
+        this.changeUserEvent.next(user);
+        this.loginEvent.next({ success: true });
+        this.router.navigate(['']);
+      },
+      err => {
+        this.loginEvent.next({ success: false, message: err });
+      }
       );
   }
 
   resetPassword(username: string) {
     this.http.post(
       this.resetPasswordUrl, { username }, { observe: 'response' })
       .subscribe(
-        response => {
-          if (response.status === 201) {
-            this.resetPasswordEvent.next({ success: true });
-          } else {
-            this.resetPasswordEvent.next(
-              {
-                success: false,
-                message: `Backend return http code other than 201: ${response.status}`
-              });
-          }
-        },
-        err => {
-          this.resetPasswordEvent.next({ success: false, message: err });
-        });
+      response => {
+        if (response.status === 201) {
+          this.resetPasswordEvent.next({ success: true });
+        } else {
+          this.resetPasswordEvent.next(
+            {
+              success: false,
+              message: `Backend return http code other than 201: ${response.status}`
+            });
+        }
+      },
+      err => {
+        this.resetPasswordEvent.next({ success: false, message: err });
+      });
   }
 
   confirmResetPassword(password: string, uidb64: string, token: string) {
     this.http.post(
       `${this.resetPasswordUrl}/${uidb64}/${token}`, { password }, { observe: 'response' })
       .subscribe(
-        response => {
-          if (response.status === 201) {
-            this.confirmResetPasswordEvent.next({ success: true });
-          } else {
-            this.confirmResetPasswordEvent.next(
-              {
-                success: false,
-                message: `Backend return http code other than 201: ${response.status}`
-              });
-          }
-        },
-        err => {
-          this.confirmResetPasswordEvent.next({ success: false, message: err });
-        });
+      response => {
+        if (response.status === 201) {
+          this.confirmResetPasswordEvent.next({ success: true });
+        } else {
+          this.confirmResetPasswordEvent.next(
+            {
+              success: false,
+              message: `Backend return http code other than 201: ${response.status}`
+            });
+        }
+      },
+      err => {
+        this.confirmResetPasswordEvent.next({ success: false, message: err });
+      });
+  }
+
+  register(email: string, password: string) {
+    return this.http.post(
+      this.registerUrl,
+      { password, email },
+      { observe: 'response' })
+  }
+
+  activateAccount(uuid: string) {
+    return this.http.post(
```

Comment:
I remember your discussion with @jacol12345 regarding one-atrtribute `accountActivationRequestModel`. Wouldn't it be better here to instead passing `uuid` in request payload, just request `this.accountActivationUrl + uuid + "/"` like:
```
activateAccount(uuid: string) {
  return this.http.post(
    `${this.accountActivationUrl}${uuid}/`,
    { observe: 'response' });
}
```
and in Django-based `urls.py` (it looks exactly the same as old, `/o/activate/...`):
```
url(
    r'^api/activate/(?P<uuid>[-0-9A-Za-z]+)/$',
    api_views.activate_view,
    name='activate
)
```
and in Django view:
```
def activate_view(request, uuid):
    """View responsible for activating user account."""
    try:
        profile = UserProfile.objects.get(uuid=uuid)
...
```